crates/store/src/state.rs

```diff
@@ -1088,9 +1119,21 @@ async fn load_mmr(db: &mut Db) -> Result<Mmr, StateInitializationError> {
 }
 
 #[instrument(level = "info", target = COMPONENT, skip_all)]
-async fn load_account_tree(db: &mut Db) -> Result<AccountTree, StateInitializationError> {
+async fn load_account_tree(
+    db: &mut Db,
+    block_number: BlockNumber,
+) -> Result<AccountTreeWithHistory<AccountTree<LargeSmt<MemoryStorage>>>, StateInitializationError>
```

Comment:
nit: we could use `AccountTreeWithHistory<InMemoryAccountTree>` instead of `AccountTreeWithHistory<AccountTree<LargeSmt<MemoryStorage>>` - right?